plugins/flatpak/gs-flatpak.c

```diff
@@ -1388,10 +1469,29 @@ gs_flatpak_add_updates_pending (GsFlatpak *self, GsAppList *list,
 			g_warning ("failed to add flatpak: %s", error_local->message);
 			continue;
 		}
-		gs_app_set_state (app, AS_APP_STATE_UPDATABLE_LIVE);
+
+		if (gs_app_get_kind (app) == AS_APP_KIND_RUNTIME)
+			main_app = get_main_app_of_related (self, app, cancellable,
+							    &error_local);
+
+		if (main_app == NULL) {
+			if (error_local != NULL &&
+			    !g_error_matches (error_local, G_IO_ERROR, G_IO_ERROR_NOT_FOUND)) {
+				g_debug ("Couldn't get the main app for related app %s: %s; continuing",
```

Comment:
The text in this `g_debug()` suggests that a `continue` is missing.

Reply:
Hmmm, not really. Why would it have to be tied to a `continue` statement? What it means is that it's not aborting the main op. This is only because in the past we've seen some scary warnings only to find out they don't skip any execution.

Reply:
Ok, in that case the text in there is a bit confusing. I'd suggest replacing "continuing" for something that better reflects what's happening right afterwards (`main_app = app`)

Reply:
That block of code was getting complex with all the error handling and comments so I split it out into a different function. Not a big fan of that because of the warning/debug messages but I think overall it's better this way. Because of that I have removed the continuing (as in the context of a helper function is not so useful anymore).